src/main/java/co/cask/hydrator/KafkaBatchSource.java

```
@@ -0,0 +1,579 @@
+
+        boolean offsetUnset = request.getOffset() == KafkaRequest.DEFAULT_OFFSET;
+        // When the offset is unset, it means it's a new topic/partition, we also need to consume the earliest offset
+        if (offsetUnset) {
+          request.setOffset(request.getEarliestOffset());
+          offsetKeys.put(
+            request,
+            new KafkaKey(request.getTopic(), request.getLeaderId(), request.getPartition(), 0, request
+              .getOffset()));
+        }
+      }
+    }
+    return finalRequests;
+  }
+
+  private List<TopicMetadata> getKafkaMetadata(List<String> metaRequestTopics) {
+    String brokerString = config.getBrokers();
+    if (brokerString.isEmpty()) {
+      throw new InvalidParameterException("kafka.brokers must contain at least one node");
+    }
+    List<String> brokers = Arrays.asList(brokerString.split("\\s*,\\s*"));
+    Collections.shuffle(brokers);
+    boolean fetchMetaDataSucceeded = false;
+    int i = 0;
+    List<TopicMetadata> topicMetadataList = null;
+    while (i < brokers.size() && !fetchMetaDataSucceeded) {
+      SimpleConsumer consumer = createBrokerConsumer(brokers.get(i));
+      LOG.info(String.format("Fetching metadata from broker %s with client id %s for %d topic(s) %s", brokers.get(i),
+                             consumer.clientId(), metaRequestTopics.size(), metaRequestTopics));
+      try {
+        topicMetadataList = consumer.send(new TopicMetadataRequest(metaRequestTopics)).topicsMetadata();
+        fetchMetaDataSucceeded = true;
+      } finally {
+        consumer.close();
+        i++;
+      }
+    }
+    return topicMetadataList;
+  }
+
+  private SimpleConsumer createBrokerConsumer(String broker) {
+    if (!broker.matches(".+:\\d+")) {
+      throw new InvalidParameterException("The kakfa broker " + broker + " must follow address:port pattern");
+    }
+    String[] hostPort = broker.split(":");
+    return createSimpleConsumer(hostPort[0], Integer.valueOf(hostPort[1]));
```

if the port is not an integer we should throw an exception with a descriptive message.